File: backend/app/schemas/amazon_ads.py
from typing import List, Optional, Dict, Any
from datetime import date, datetime
from uuid import UUID
from pydantic import BaseModel

# --- Existing Schemas (Restored) ---

class AmazonStoreSchema(BaseModel):
    id: UUID
    store_name: str
    marketplace_name: str
    currency_code: str
    
    class Config:
        from_attributes = True

class AdsMatrixPoint(BaseModel):
    sku: str
    asin: str
    status: str  # 'healthy', 'risk', 'opportunity'
    stock_weeks: float
    tacos: float
    sales: float
    ad_spend: float
    ctr: float
    cvr: float
    acos: float
    roas: float
    margin: float

class AdsDiagnosis(BaseModel):
    sku: str
    asin: str
    status: str
    diagnosis: str
    metrics: Dict[str, Any]

# --- New Campaign Schemas ---

class CampaignBase(BaseModel):
    name: str
    state: str
    campaign_type: str
    targeting_type: str
    daily_budget: float
    start_date: Optional[date] = None
    end_date: Optional[date] = None

class CampaignMetrics(BaseModel):
    impressions: int = 0
    clicks: int = 0
    spend: float = 0.0
    sales: float = 0.0
    orders: int = 0
    units: int = 0
    
    # Derived metrics
    ctr: float = 0.0
    cpc: float = 0.0
    acos: float = 0.0
    roas: float = 0.0
    cvr: float = 0.0

class CampaignDetail(CampaignBase, CampaignMetrics):
    id: UUID
    campaign_id: str # Amazon ID

    class Config:
        from_attributes = True

class CampaignListResponse(BaseModel):
    total: int
    items: List[CampaignDetail]
    page: int
    limit: int

# --- Diagnosis Schemas ---

class WastedSpendCampaign(BaseModel):
    id: str
    name: str
    spend: float
    clicks: int
    orders: int

class WastedSpendResponse(BaseModel):
    total_wasted_spend: float
    campaign_count: int
    campaigns: List[WastedSpendCampaign]

class HighAcosCampaign(BaseModel):
    id: str
    name: str
    spend: float
    sales: float
    acos: float
    target_acos: float

class HighAcosResponse(BaseModel):
    campaign_count: int
    campaigns: List[HighAcosCampaign]

File: frontend/src/api/amazon-ads.ts
import { http } from '@/utils/http'

export interface Campaign {
    id: string
    campaign_id: string
    name: string
    state: string
    campaign_type: string
    targeting_type: string
    daily_budget: number
    start_date: string
    end_date: string | null

    // Metrics
    impressions: number
    clicks: number
    spend: number
    sales: number
    orders: number
    units: number
    ctr: number
    cpc: number
    acos: number
    roas: number
    cvr: number
}

export interface CampaignListResponse {
    total: number
    items: Campaign[]
    page: number
    limit: number
}

export interface CampaignQueryParams {
    store_id?: string
    state?: string
    campaign_type?: string
    start_date?: string
    end_date?: string
    page?: number
    limit?: number
    sort_by?: string
    sort_order?: 'asc' | 'desc'
}

export interface WastedSpendCampaign {
    id: string
    name: string
    spend: number
    clicks: number
    orders: number
}

export interface WastedSpendResponse {
    total_wasted_spend: number
    campaign_count: number
    campaigns: WastedSpendCampaign[]
}

export interface HighAcosCampaign {
    id: string
    name: string
    spend: number
    sales: number
    acos: number
    target_acos: number
}

export interface HighAcosResponse {
    campaign_count: number
    campaigns: HighAcosCampaign[]
}

export const amazonAdsApi = {
    getCampaigns: (params: CampaignQueryParams) => {
        return http.get<CampaignListResponse>('/amazon/campaigns', { params })
    },
    getWastedSpend: (storeId: string, days: number = 7, threshold: number = 50) => {
        return http.get<WastedSpendResponse>('/ads-analysis/diagnosis/wasted-spend', {
            params: { store_id: storeId, days, threshold }
        })
    },
    getHighAcos: (storeId: string, days: number = 7, acosThreshold: number = 30, minSpend: number = 50) => {
        return http.get<HighAcosResponse>('/ads-analysis/diagnosis/high-acos', {
            params: { store_id: storeId, days, acos_threshold: acosThreshold, min_spend: minSpend }
        })
    }
}
